From 5bc338da693cb7590b9b039548b0624fd244d311 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Thu, 30 Dec 2021 00:24:32 +0100
Subject: [PATCH 1/2] make updates page default at startup

This is mostly to avoid having the startup of an application
show a window that looks broken. It should be reverted once
the explore page is fully adaptive and filled with beautiful
data.

Simply modifying the desktop file in phosh is not enough because
the application is dbus activated. In consequence, the Exec
in the desktop is ignored. Making the activation accept parameters
might be possible but out of scope
---
 src/gs-application.c              | 2 +-
 src/org.gnome.Software.desktop.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gs-application.c b/src/gs-application.c
index 90487df6..d35edc1c 100644
--- a/src/gs-application.c
+++ b/src/gs-application.c
@@ -1064,7 +1064,7 @@ gs_application_activate (GApplication *application)
 	GsApplication *app = GS_APPLICATION (application);
 
 	if (app->shell_loaded_handler_id == 0)
-		gs_shell_set_mode (app->shell, GS_SHELL_MODE_OVERVIEW);
+		gs_shell_set_mode (app->shell, GS_SHELL_MODE_UPDATES);
 
 	gs_shell_activate (GS_APPLICATION (application)->shell);
 
diff --git a/src/org.gnome.Software.desktop.in b/src/org.gnome.Software.desktop.in
index 318c243a..b79666eb 100644
--- a/src/org.gnome.Software.desktop.in
+++ b/src/org.gnome.Software.desktop.in
@@ -3,7 +3,7 @@ Name=Software
 Comment=Add, remove or update software on this computer
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
 Icon=@application_id@
-Exec=gnome-software %U
+Exec=gnome-software --mode=updates %U
 Terminal=false
 Type=Application
 Categories=GNOME;GTK;System;PackageManager;
-- 
2.25.1

